src/Device/Port/TTYEnumerator.cpp

```diff
@@ -32,26 +32,25 @@ Copyright_License {
 static bool
 CheckTTYName(const char *name) noexcept
 {
-  /* filter "/dev/tty*" */
-  if (const char *t = StringAfterPrefix(name, "tty"); t != nullptr) {
-    if (*t == 0)
-      /* ignore /dev/tty */
-      return false;
-
-    /* ignore virtual internal ports on Mac OS X (and probably other
-       BSDs) */
-    if (*t >= 'p' && *t <= 'w')
-      return false;
-
-    /* filter out "/dev/tty0", ... (valid integer after "tty") */
-    if (IsDigitASCII(*t))
-      return false;
-
+  /*
+   * rfcomm devices do not have a 'device/driver' file that is used
+   * to exclude virtual/pseudo terminals
+   */
+  if (StringStartsWith(name, "rfcomm"))
     return true;
-  } else if (StringStartsWith(name, "rfcomm"))
+
+  /*
+   * only entries with existing 'device/driver' folder
+   * aren't virtual/pseudo terminals
+   */
+  char path[80];
+  sprintf(path, "/sys/class/tty/%s/device/driver", name);
+  DIR* dir = opendir(path);
+  if (dir) {
+    closedir(dir);
```

Comment:
What's the point of opening a file descriptor and creating the DIR object when all you want to know is whether the directory exists?